src/components/views/settings/KeyBackupPanel.js

```diff
@@ -0,0 +1,240 @@
+                let sigStat;
+                if (sig.device.getFingerprint() === MatrixClientPeg.get().getDeviceEd25519Key()) {
+                    sigStat = _t(
+                        "Backup has a <validity>valid</validity> signature from this device",
+                        {}, sigStatSub,
+                    );
+                } else if (sig.valid && sig.device.isVerified()) {
+                    sigStat = _t(
+                        "Backup has a <validity>valid</validity> signature from " +
+                        "<verify>verified</verify> device <device>x</device>",
+                        {}, sigStatSub,
+                    );
+                } else if (sig.valid && !sig.device.isVerified()) {
+                    sigStat = _t(
+                        "Backup has a <validity>valid</validity> signature from " +
+                        "<verify>unverified</verify> device <device></device>",
+                        {}, sigStatSub,
+                    );
+                } else if (!sig.valid && sig.device.isVerified()) {
+                    sigStat = _t(
+                        "Backup has an <validity>invalid</validity> signature from " +
+                        "<verify>verified</verify> device <device></device>",
+                        {}, sigStatSub,
+                    );
+                } else if (!sig.valid && !sig.device.isVerified()) {
+                    sigStat = _t(
+                        "Backup has an <validity>invalid</validity> signature from " +
+                        "<verify>unverified</verify> device <device></device>",
+                        {}, sigStatSub,
+                    );
+                }
+
+                let verifyButton;
+                if (!sig.device.isVerified()) {
+                    verifyButton = <div><br /><AccessibleButton className="mx_UserSettings_button"
+                            onClick={this._verifyDevice} data-sigindex={i}>
+                        { _t("Verify...") }
+                    </AccessibleButton></div>;
+                }
+
+                return <div key={i}>
+                    {sigStat}
+                    {verifyButton}
+                </div>;
+            });
+            if (this.state.backupSigStatus.sigs.length === 0) {
+                backupSigStatuses = _t("Backup is not signed by any of your devices");
+            }
+
+            return <div>
+                {_t("Backup version: ")}{this.state.backupInfo.version}<br />
+                {_t("Algorithm: ")}{this.state.backupInfo.algorithm}<br />
+                {clientBackupStatus}<br />
+                <div>{backupSigStatuses}</div><br />
+                <br />
+                <AccessibleButton className="mx_UserSettings_button"
+                        onClick={this._restoreBackup}>
+                    { _t("Restore backup") }
+                </AccessibleButton>&nbsp;&nbsp;&nbsp;
+                <AccessibleButton className="mx_UserSettings_button danger"
```

Comment:
we have a top level CSS class called `danger`? what happened to namespacing?

Reply:
mm, yeah, this should be namespaced really. maybe one to fix in the redesign?